src/expr.rs

```diff
@@ -2101,6 +2101,25 @@ pub(crate) fn rewrite_assign_rhs_expr<R: Rewrite>(
     rhs_kind: &RhsAssignKind<'_>,
     rhs_tactics: RhsTactics,
 ) -> Option<String> {
+    let get_lhs_last_line_shape = || {
+        let last_line = lhs.rsplitn(2, "\n").next().unwrap_or_default();
```

Comment:
This would be much easier to understand:
```suggestion
        let last_line = lhs.lines().last().unwrap_or_default();
```